cpp/oneapi/dal/algo/basic_statistics/test/online_spmd.cpp

```diff
@@ -0,0 +1,161 @@
+        std::vector<partial_result_t> partial_results;
+
+        auto input_table = base_t::template split_table_by_rows<double>(data, rank_count_);
+        if (use_weights) {
+            auto input_weights = base_t::template split_table_by_rows<double>(weights, rank_count_);
+            for (int64_t i = 0; i < rank_count_; i++) {
+                dal::basic_statistics::partial_compute_result<> partial_result;
+                auto input_table_blocks =
+                    base_t::template split_table_by_rows<double>(input_table[i], blocks_count_);
+                auto input_weights_blocks =
+                    base_t::template split_table_by_rows<double>(input_weights[i], blocks_count_);
+                for (int64_t j = 0; j < blocks_count_; j++) {
+                    partial_result = this->partial_compute(bs_desc,
+                                                           partial_result,
+                                                           input_table_blocks[j],
+                                                           input_weights_blocks[j]);
+                }
+                partial_results.push_back(partial_result);
+            }
+            const auto compute_result = this->finalize_compute_override(bs_desc, partial_results);
+
+            base_t::check_compute_result(compute_mode, data, weights, compute_result);
+        }
+        else {
+            for (int64_t i = 0; i < rank_count_; i++) {
+                dal::basic_statistics::partial_compute_result<> partial_result;
+                auto input_table_blocks =
+                    base_t::template split_table_by_rows<double>(input_table[i], blocks_count_);
+                for (int64_t j = 0; j < blocks_count_; j++) {
+                    partial_result =
+                        this->partial_compute(bs_desc, partial_result, input_table_blocks[j]);
+                }
+                partial_results.push_back(partial_result);
+            }
+            const auto compute_result = this->finalize_compute_override(bs_desc, partial_results);
+
+            base_t::check_compute_result(compute_mode, data, table{}, compute_result);
+        }
+    }
+
+private:
+    std::int64_t rank_count_;
+    std::int64_t blocks_count_;
+};
+
+using basic_statistics_types = COMBINE_TYPES((float, double), (basic_statistics::method::dense));
+
+TEMPLATE_LIST_TEST_M(basic_statistics_online_spmd_test,
+                     "basic_statistics common flow",
+                     "[basic_statistics][integration][spmd]",
+                     basic_statistics_types) {
+    SKIP_IF(this->get_policy().is_cpu());
+    SKIP_IF(this->not_float64_friendly());
+
+    const te::dataframe data =
+        GENERATE_DATAFRAME(te::dataframe_builder{ 1000, 100 }.fill_normal(-30, 30, 7777),
+                           te::dataframe_builder{ 2000, 20 }.fill_normal(0, 1, 7777),
+                           te::dataframe_builder{ 2500, 20 }.fill_normal(-30, 30, 7777));
```

Comment:
The datasets sizes are really tiny considering online_spmd scenario. I think it worth to increase row counts and column counts.